File: 链式编程(计算器)/链式编程(计算器)/ViewController.m
//
//  ViewController.m
//  链式编程(计算器)
//
//  Created by bcmac3 on 16/4/8.
//  Copyright © 2016年 KellenYangs. All rights reserved.
//

#import "ViewController.h"
#import "NSObject+Caculator.h"
#import "CaculatorMaker.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    int num =  [NSObject makeCaculator:^(CaculatorMaker *maker){
        maker.add(10);
        maker.add(20);
    }];
    NSLog(@"%zd", num);
    
    CaculatorMaker *maker = [[CaculatorMaker alloc] init];
    NSLog(@"%ld", [maker.add(10).add(20) result]);
    
   
    
}

@end

File: KLEmojiViewExample/KLEmojiViewExample/ViewController.swift
//
//  ViewController.swift
//  KLEmojiViewExample
//
//  Created by bcmac3 on 16/4/6.
//  Copyright © 2016年 KellenYangs. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var textView: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        addChildViewController(emotionVC)
        
        textView.inputView = emotionVC.view
    }
    
    // MARK: -- 懒加载
    private lazy var emotionVC: KLEmoticonViewController = KLEmoticonViewController { [weak self] (emoticon) -> () in
        print(emoticon.chs)
        self!.textView.insertEmoticon(emoticon)
    }

}

